src/scripts/users/role-manager.js

```diff
@@ -0,0 +1,181 @@
+const { MessageMenuOption, MessageMenu, MessageActionRow } = require('discord-buttons');
+
+const COMMAND = "!role message";
+const EMOJI_ROLE_ID_DELIMITER = ':';
+const ROLE_ID_DESCRIPTION_DELIMITER = '-';
+const ROLE_SELECTION_MESSAGE = "Select your roles here!";
+
+function handleRoleMessage(message) {
+    if (!isRoleSelectCreationMessage(message)) {
+        return;
+    }
+    if (!messageAuthorIsMod(message)) {
+        return;
+    }
+    const roleOptionStrings = getLinesWhichContainRoleOptionData(message);
+    if (!roleOptionStrings) {
+        return;
+    }
+    const roleOptions = createRoleOptionsArray(message, roleOptionStrings);
+    if (!roleOptions || roleOptions.length === 0) {
+        return;
+    }
+
+    createRoleOptionsForMessage(message, roleOptions);
+}
+
+function isRoleSelectCreationMessage(message) {
+    return message.content.toLowerCase().startsWith(COMMAND);
+}
+
+function messageAuthorIsMod(message) {
+    return Boolean(message.member.roles.cache.get(process.env.MODERATORS));
+}
+
+function getLinesWhichContainRoleOptionData(message) {
+    return message.content.split('\n').filter(function (line) {
+        const validRoleOptionRegex = /[\p{Emoji}]{1}.*:.*\d+/u;
+        return validRoleOptionRegex.test(line);
+    });
+}
```

Comment:
If you pull the regex out of the filter, you can return a beautiful one-line function :D 
```js
function getLinesWhichContainRoleOptionData(message) {
    const validRoleOptionRegex = /[\p{Emoji}]{1}.*:.*\d+/u;
    return message.content.split('\n').filter((line) => validRoleOptionRegex.test(line));
}
```
Functionnal alternative
```js
function getLinesWhichContainRoleOptionData(message) {
    const validRoleOptionRegex = /[\p{Emoji}]{1}.*:.*\d+/u;
    const testValidRoleOptionRegex = (line) => validRoleOptionRegex.test(line);
    return message.content.split('\n').filter(testValidRoleOptionRegex);
}
```
Or

```js
function getLinesWhichContainRoleOptionData(message) {
    const testValidRoleOptionRegex = (line) => /[\p{Emoji}]{1}.*:.*\d+/u.test(line);
    return message.content.split('\n').filter(testValidRoleOptionRegex);
}
```